File: backend/src/Sixram.Nextgen.JinmarxWellness.Domain/Entities/CameraEvent.cs
using Sixram.Nextgen.JinmarxWellness.Domain.Common;

namespace Sixram.Nextgen.JinmarxWellness.Domain.Entities;

public class CameraEvent : AuditableEntity
{
    public string? CameraIp { get; set; }
    public string? Ipv6Address { get; set; }
    public int? PortNo { get; set; }
    public string? Protocol { get; set; }
    public string? MacAddress { get; set; }
    public int? ChannelId { get; set; }
    public string? ChannelName { get; set; }
    public DateTimeOffset EventDateTime { get; set; }
    public string EventType { get; set; } = string.Empty;
    public string EventState { get; set; } = string.Empty;
    public string? EventDescription { get; set; }
    public int? ActivePostCount { get; set; }
    public string? Source { get; set; }
    public string? SnapshotPath { get; set; }
    public string? SnapshotUrl { get; set; }
    public DateTimeOffset? SnapshotCapturedAt { get; set; }
    public string? SnapshotError { get; set; }
    public string RawXml { get; set; } = string.Empty;
}

File: app/src/types/cameraEvent.ts
export type CameraEventRow = {
  id: number;
  cameraIp?: string | null;
  channelId?: number | null;
  channelName?: string | null;
  eventDateTime: string;
  eventType: string;
  eventState: string;
  eventDescription?: string | null;
  activePostCount?: number | null;
  source?: string | null;
  snapshotUrl?: string | null;
  snapshotCapturedAt?: string | null;
  snapshotError?: string | null;
  createdAt?: string | null;
};

export type CameraEventSummary = {
  totalToday: number;
  activeToday: number;
  vmdActiveToday: number;
  total: number;
  active: number;
  vmdActive: number;
  captured: number;
  snapshotFailed: number;
  lastEvent?: CameraEventRow | null;
};

export type CameraEventQuery = {
  take?: number;
  eventType?: string;
  eventState?: string;
  start?: string;
  end?: string;
  capturedStart?: string;
  capturedEnd?: string;
};

export type CameraEventSettings = {
  cameraEventCooldownSeconds: number;
  snapshotCaptureEnabled: boolean;
  cameraIp: string;
  username: string;
  snapshotChannel: string;
  passwordConfigured: boolean;
};

export type UpdateHikvisionCameraSettingsRequest = {
  cameraEventCooldownSeconds?: number;
  snapshotCaptureEnabled?: boolean;
  cameraIp?: string;
  username?: string;
  password?: string;
  snapshotChannel?: string;
};

export type ClearCameraEventsResult = {
  deletedCount: number;
};

export type CameraSnapshotTestResult = {
  capturedAt: string;
  imageDataUrl: string;
};
